relay_messages_test.go

```diff
@@ -160,6 +160,20 @@ func withLazyCallReqCombinations(f func(cr testCallReq)) {
 
 type testCallRes int
 
+type testCallResParams struct {
+	payloadSize       int
```

Comment:
not really sure how `payloadSize` interacts with all the other fields (e.g., what happens if the length of arg1+2+3 > payloadSize).

`testCallReqParams` does not have a payload size, not sure if we're intentionally doing something different for call res construction vs call req construction

Reply:
this is useful for creating corrupted frames that trigger a buffer read error, and is used in a test in the next diff.

Reply:
Perhaps it should be a `if payloadSize > 0` then write it out as the payload size in the header?

with the current implementation,
 - if payloadSize is too small, then I think the typed writer will start to fail (which we check with `require.NoError`
 - if payloadSize is too large, then it has no impact, since we overwrite the frame header's payload size with bytes written.

Perhaps I'm missing something, not sure how it'll work with the next diff

Reply:
yea i suppose it's not really necessary - it would make the numbers smaller and easier to reason about, so we can trigger an error with e.g.
```
	_, err := newLazyCallRes(newCallResFrame(t, testCallResParams{
		payloadSize: 100,
		arg2Prefix:  []byte{0, 100},
		arg2KeyVals: exampleArg2Map,
	}))
```
but it's not strictly necessary. we should be able to achieve something similar with a larger number in `arg2Prefix`. will change in #830